torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -839,6 +852,17 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
   bool storeError_{false};
 
+  // The lock which protects the write/read of
+  // extendedTimeout_/inflightTimeoutExt_.
+  std::mutex mtxTimeoutExtension_;
```

Comment:
As you already have a mutex

```
  std::mutex mutex_;
```

You really need to explain why you need a second one, since two mutexes opens up the possibility for deadlock. It's not free!

Reply:
Honestly `std::mutex mutex_;` steals the generic name..  It is mostly used when it comes to exception handling and `devNCCLCommMap_`, etc...

OK, I think we might need to clean up and try to control number of mutex here. We can reuse that mutex_ instead of creating a new one.

Reply:
Wait, this `mutex_` is used when doing NCCLCommAbort which can have hang...We probably cannot use it. But I agree, we need to do an audit to the mutex here and see if we can consolidate them.